pkg/api/validation/validation.go

```diff
@@ -2575,6 +2575,10 @@ func ValidateServiceUpdate(service, oldService *api.Service) field.ErrorList {
 		allErrs = append(allErrs, ValidateImmutableField(service.Spec.ClusterIP, oldService.Spec.ClusterIP, field.NewPath("spec", "clusterIP"))...)
 	}
 
+	if oldService.Spec.ClusterIP == api.ClusterIPNone {
+		allErrs = append(allErrs, ValidateImmutableField(service.Spec.ClusterIP, oldService.Spec.ClusterIP, field.NewPath("spec", "clusterIP"))...)
```

Comment:
oh, good point.

OK, can we then make this code more obvious and comment it heavily to explain which transitions are being caught and which are not?